packages/contracts/test/registry/dao-registry.ts

```diff
@@ -140,4 +141,35 @@ describe('DAORegistry', function () {
       .to.be.revertedWithCustomError(ensSubdomainRegistrar, 'AlreadyRegistered')
       .withArgs(daoDomainHash, ensSubdomainRegistrar.address);
   });
+
+  it('should revert if name contains any invalid char', async () => {
+    const baseName = 'this-is-my-super-valid-name';
+
+    // loop through the ascii table
+    for (let i = 0; i < 127; i++) {
+      const newTargetDao = await deployNewDAO(ownerAddress);
+      // random place for the char to improve validation check
+      const placement = Math.round(Math.random() * baseName.length);
+      const daoName =
+        baseName.substring(0, placement) +
+        String.fromCharCode(i) +
+        baseName.substring(placement + 1);
+
+      // test success if it is a valid char [0-9a-z\-]
+      if ((i > 47 && i < 58) || (i > 96 && i < 123) || i === 45) {
+        await expect(
+          daoRegistry.register(newTargetDao.address, ownerAddress, daoName)
```

Comment:
1. Once, free level function is created, we should be having tests for it.
2.  Does this test include every case ? Seems like not.  As all it tests is `daoName` not being registered after it's been registered ? couple of more tests required to include upper case letter tests , special chars and etc.

Reply:
we have tests now but we have to repeat the tests in both registries because we don't have a mocking system in place and we need to make sure that these functions work as expected